---
title: "Making_third_plots"
author: "DLS"
date: "2023-12-18"
output: html_document
---

# Load Packages
```{r}
library(tidyverse)
library(reshape2)
library(cowplot)
library(Rmisc)
```

# Load in data + calc rela abund
```{r}
# Read in vOTU table
vOTU_table = readRDS("/Users/dinosbardellati/Desktop/PhD_stuff/projects/Bee_phage/5kb/reviewer_comments/coverage_table/5kb_phage_normalized.RDS")
# Read in meta table + clean a bit
meta = read.csv("/Users/dinosbardellati/Desktop/PhD_stuff/projects/Bee_phage/5kb/reviewer_comments/coverage_table/Bee_phage_meta.csv", header = T)
## Drop negatives
meta_virome = meta[meta$Sample %in% rownames(vOTU_table),]

# Reorder vOTU tablt to match meta
phage_df_order = vOTU_table[order(match(rownames(vOTU_table), meta_virome$Sample)),]
# Then use read counts in meta data to normalize vOTU
phage_df_norm = (phage_df_order/meta_virome$filtered_read_count)*1000000
# Lastly, convert vOTU table to relative abundances
phage_df_rela = (phage_df_norm/rowSums(phage_df_norm))*100
```

# Host taxonomy
```{r}
##### Make host assignment object ##### 
# Load in phage host_df
phage_host = read.csv("/Users/dinosbardellati/Desktop/PhD_stuff/projects/Bee_phage/5kb/cleaned_code/assembly/network/all_phage_cluster_hosts_tax.csv")
# Make al ist of all vOTU names
vOTU_phage = data.frame("vOTU_phage" = colnames(vOTU_table))

# Filter phage_host df to only those phage which occur in vOTU table
### (removes "known" phage and some phage which didnt make it into final vOTU table)
phage_host_clean = phage_host %>%
  filter(phage %in% vOTU_phage$vOTU_phage)

## Merge the two dfs together
vOTU_phage_host = vOTU_phage %>% 
  left_join(phage_host_clean, join_by("vOTU_phage" == "phage")) %>%
  replace_na(list(host = "unassigned")) %>% # if a vOTU phage is not present in phage_host_clean, make its host "unassigned"
  replace_na(list(tax = "unassigned")) # repeat with the tax col
dim(vOTU_phage_host)
##### Done ##### 

##### Add meta data and host assignment to vOTU relative abundance table ##### 
# Add sample as col to vOTU rel abund table
phage_df_clean = phage_df_rela
phage_df_clean$sample = rownames(phage_df_clean)
phage_df_clean$sample = as.factor(phage_df_clean$sample)

abund_meta_host = meta %>%
  filter(Type != "Negative") %>%
  dplyr::select(Sample, Hive_ID, Type, Bee, clean_names) %>%
  left_join(phage_df_clean, join_by("Sample" == "sample")) %>%
  melt(id = c("Sample", "Hive_ID", "Type", "Bee", "clean_names")) %>%
  left_join(vOTU_phage_host, join_by("variable" == "vOTU_phage"))

# Make a new df which calculates the sumative abundance of each phage targeting group in each sample
## (i.e. sum the relative abundance of all lactobacillus targeting phage in a single sample)
host_sum = data.frame(abund_meta_host %>%
  dplyr::group_by(Sample, Bee, Type, , Hive_ID, host, clean_names) %>%
  dplyr::summarise(sum = sum(value)))
##### Done ##### 

##### Find most abnund host targeting group ##### 
# Phage targeting which groups of bacteria are the most abundant?
## (this is sum across whole df)
abund_meta_host %>%
  group_by(host) %>%
  dplyr::summarize(sum = sum(value, na.rm=TRUE)) %>%
  arrange(desc(sum))
# Lets pull the top 7 hosts from this (after this it drops from ~10% to 1% accross the entire df)
top = c("unassigned",        
"Lactobacillus",     
"Gilliamella",       
"Bifidobacterium",
"Bombilactobacillus",
"Schmidhempelia",    
"Acinetobacter")
# Now we will make an "other" catagory
## Any phage which does not target a bacteria in this "top" bin will be tossed into the "other" category
host_sum$host_2 = ifelse(host_sum$host %in% top, host_sum$host, "other")
##### Done ##### 

##### Now we are ready to plot ##### 
# I lied, first we coerce some characters into factors, make a col to order stacked bars on, and then fix order of host assignments
host_sum_fix = host_sum %>%
  mutate(Type = factor(Type, levels=c("Virome", "Total"))) %>%
  mutate(Bee = factor(Bee, levels=c("Honey bee", "Bumble bee"))) %>%
  mutate(Order = as.numeric(str_split_fixed(Sample, "_", 3)[,3])) %>%
  mutate(host_2 = factor(host_2, levels=c("Acinetobacter",
                                          "Lactobacillus",     
                                          "Gilliamella",       
                                          "Bifidobacterium",
                                          "Bombilactobacillus",
                                          "Schmidhempelia",
                                          "other",
                                          "unassigned")))

# Make a new pal to color by
stacked_pal = palette(c("#FDAB5E",
                        "#8B9DD5",          
                        "#64C5B3",
                        "#8BD48E",
                        "#C57777",
                        "#DB8CFD",
                        "#C593AF",
                        "#7D807E"))

# Now we plot!
stacked_host_tax = ggplot(host_sum_fix, aes(x = reorder(clean_names, Order), y = sum, fill = host_2)) + 
  geom_bar(stat = "identity") +
  facet_wrap(~Bee+Type, scales = "free_x") +
  scale_fill_manual(values = stacked_pal) + 
  xlab("") +
  ylab("Relative abundance") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, color = "black", size = 15),
        strip.text = element_text(size = 20, color = "black")) 

legend = data.frame(host = c("Acinetobacter",
                      "Lactobacillus",     
                      "Gilliamella",       
                      "Bifidobacterium",
                      "Bombilactobacillus",
                      "Schmidhempelia",
                      "Other",
                      "Unassigned",
                      "Known"))

legend$host = factor(legend$host, levels = c("Acinetobacter",
                      "Lactobacillus",     
                      "Gilliamella",       
                      "Bifidobacterium",
                      "Bombilactobacillus",
                      "Schmidhempelia",
                      "Other",
                      "Unassigned",
                      "Known"))

legend_pal = palette(c("#FDAB5E",
                        "#8B9DD5",          
                        "#64C5B3",
                        "#8BD48E",
                        "#C57777",
                        "#DB8CFD",
                        "#C593AF",
                        "#7D807E",
                        "#CCCCCC"))

legend_plot = ggplot(legend, aes(x = ordered(host), 1, col = host)) +
  geom_point(size = 5) +
  scale_color_manual(values = legend_pal) +
  theme_bw() +
  theme(text = element_text(size = 15)) + 
  guides(col=guide_legend(title="Host"))
library(cowplot)

plot_grid(get_legend(legend_plot), stacked_host_tax + theme(legend.position = "none"),
          rel_widths = c(5, 10),
          rel_heights =c (10, 5))
##### Done ##### 
```

# Write data to be exported as table 
```{r}
# this is the df which has the data we are interested in
head(host_sum_fix)
# We want to aggregated this df based on the interaction of Bee_Type and host_2
## (i.e. what is the abundance of these different taxa in honey bee total metagenomes vs honey bee viromes)
# add new col with the interaction term
host_sum_fix_new = host_sum_fix %>% 
  mutate(Bee_type = paste(host_sum_fix$Bee, host_sum_fix$Type))
library(Rmisc)
host_sum_tab = summarySE(data = host_sum_fix_new, 
          measurevar = "sum", 
          groupvars = c("Bee_type", "host_2"))
write.csv(x = host_sum_tab, file = "/Users/dinosbardellati/Desktop/PhD_stuff/projects/Bee_phage/5kb/cleaned_code/third_plots/phage_host_tab.csv")

library(lme4)
library(lmerTest)
library(emmeans)
# Acineto
acineto = lmer(sum ~ Bee*Type + (1|Hive_ID), 
             data = host_sum_fix_new,
             subset = host == "Acinetobacter")
summary(acineto)
x = emmeans(acineto, c("Bee", "Type"))
pairs(x, adjust="bonferroni")

# Lacto
lacto = lmer(sum ~ Bee*Type + (1|Hive_ID), 
             data = host_sum_fix_new,
             subset = host == "Lactobacillus")
summary(lacto)
x = emmeans(lacto, c("Bee", "Type"))
pairs(x, adjust="bonferroni")

# Gilli
gilli = lmer(sum ~ Bee*Type + (1|Hive_ID), 
             data = host_sum_fix_new,
             subset = host == "Gilliamella")
summary(gilli)
x = emmeans(gilli, c("Bee", "Type"))
pairs(x, adjust="bonferroni")

# Bifido
bifido = lmer(sum ~ Bee*Type + (1|Hive_ID), 
             data = host_sum_fix_new,
             subset = host == "Bifidobacterium")
summary(bifido)
x = emmeans(bifido, c("Bee", "Type"))
pairs(x, adjust="bonferroni")

# Bombilacto
bombi = lmer(sum ~ Bee*Type + (1|Hive_ID), 
             data = host_sum_fix_new,
             subset = host == "Bombilactobacillus")
summary(bombi)
# no sig

# Schmid
schmid = lmer(sum ~ Bee*Type + (1|Hive_ID), 
             data = host_sum_fix_new,
             subset = host == "Schmidhempelia")
summary(schmid)
x = emmeans(schmid, c("Bee", "Type"))
pairs(x, adjust="bonferroni")
```

# Phage taxonomy plot (supplemental)
```{r}
##### Make a plotable df #####
# Drop unassigned
vOTU_tax = vOTU_phage_host[!vOTU_phage_host$tax %in% c("unassigned", "mixed", "NA"),]
# set levels
vOTU_tax$tax = factor(vOTU_tax$tax , levels = c("Zobellviridae",
                                                "Autographiviridae",
                                                "Podoviridae",
                                                "Myoviridae",
                                                "Siphoviridae"))

tax_pal = palette(c("#E69F00",
                    "#56B4E9",
                    "#009E73",
                    "#D55E00",
                    "#0072B2"))
# plot it
phage_tax = ggplot(vOTU_tax, aes(tax, fill = tax)) +
  geom_bar() +
  scale_fill_manual(values = tax_pal) +
  theme_minimal() +
  xlab("") +
  ylab("Frequency") +
  theme(plot.title = element_text(hjust = 0.5), 
        text = element_text(size = 25), 
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
        legend.position = "right")


tax_legend = plot_grid(get_legend(ggplot(vOTU_tax, aes(tax, fill = tax)) +
  geom_bar() +
  theme_minimal() +
  xlab("Phage genus") +
  ylab("Frequency") +
  scale_fill_manual("Phage Family", values = tax_pal) + 
  xlab("") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5), 
        text = element_text(size = 25), 
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
        legend.position = "left")))




# We've already done a lot of the leg work for this!
## get sum of different groups
# Make a new df which calculates the sumative abundance of each phage targeting group in each sample
## (i.e. sum the relative abundance of all lactobacillus targeting phage in a single sample)
tax_sum = data.frame(abund_meta_host %>%
  dplyr::group_by(Sample, Bee, Type, tax) %>%
  dplyr::summarise(sum = sum(value))) %>%
  mutate(Type = factor(Type, levels=c("Virome", "Total"))) %>%
  mutate(Bee = factor(Bee, levels=c("Honey bee", "Bumble bee"))) %>%
  mutate(Order = as.numeric(str_split_fixed(Sample, "_", 3)[,3]))

ggplot(tax_sum, aes(x = reorder(Sample, Order), y = sum, fill = tax)) + 
  geom_bar(stat = "identity") +
  facet_wrap(~Bee+Type, scales = "free_x") +
  xlab("") +
  ylab("Relative abundance") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5), 
        text = element_text(size = 25), 
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
        legend.position = "none")
##### Done ##### 



# Have to add temperate phage to our network object too!
## Network .csv is here
network = read.csv("/Users/dinosbardellati/Desktop/PhD_stuff/projects/Bee_phage/5kb/network/all_phage_final_network_lifestyle.csv")
network$tax_study = paste(network$tax, network$Study, sep = "_")
# write it out as new df
# write.csv(network, "/Users/dinosbardellati/Desktop/PhD_stuff/projects/Bee_phage/5kb/network/all_phage_final_network_fixed.csv",
#       quote = F)
```
