{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c283b8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now that we have vOTUs, we can do all sorts of stuff!\n",
    "# For now, lets take our vOTUs and compare them to previously described bee phage\n",
    "# To do this, we will use...\n",
    "    # Prokka to annotate our phage seqs (we need the .faa files this tool generates for the next tools)\n",
    "    # vContact2 to construct gene sharing networks\n",
    "    # clinker to make nice multi genome alignment plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af044787",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Annotate phage with prokka\n",
    "# Once we have all of our phage, plus previously described phage in one place, we will annotate them using prokka\n",
    "# This is an example of the script used to do the annotation\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=prokka_%j\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 3:00:00\n",
    "#SBATCH --ntasks=6\n",
    "#SBATCH --output=/home/dsbard/bee_phage/reviewer_approach/annot/log/prokka%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "conda activate prokka\n",
    "# module load ncbi-blast\n",
    "\n",
    "prokka --kingdom Bacteria $1  \\\n",
    "--outdir /home/dsbard/bee_phage/reviewer_approach/annot/annotations/${1%%.fa} \\\n",
    "--hmms /home/dsbard/phrogs/all_phrogs.hmm \\\n",
    "--prefix annotate_${1%%.fa}\n",
    "\n",
    "# we run this from a directory which contains all phage (known and novel) as seperate files\n",
    "## /home/dsbard/bee_phage/reviewer_approach/drep/dereplicated_phage/dereplicated_genomes\n",
    "# This script is run with a forloop (like our previous ones)\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/annot/all_phage\n",
    "for f in *.fa\n",
    "do\n",
    "sbatch ../../scripts/annot.sh $f\n",
    "done\n",
    "\n",
    "# The only additional thing to note here is that we are using Phrogs hidden markov model db to annotate\n",
    "# This is the website: https://phrogs.lmge.uca.fr/\n",
    "# The paper: https://academic.oup.com/nargab/article/3/3/lqab067/6342220?login=false\n",
    "# The actual .hmm I used was formated for Prokka use by the Millardlab and can be found here: https://millardlab.org/2021/11/21/phage-annotation-with-phrogs/\n",
    "# A big thanks to all these people for making this available!!!\n",
    "\n",
    "# Once we have run the genomes through prokka, we can pull their .faa files\n",
    "for f in */*.faa\n",
    "do \n",
    "cp $f ../faa_files/\n",
    "done\n",
    "# For whatever reason, prokka doesnt always get rid of intermediate files\n",
    "## remove these \n",
    "rm *tmp*\n",
    "# Additionally, for whatever reason, 4 of the phage identified in busby et al do not get annotated (produce empty files)\n",
    "## we can find these with \n",
    "find . -size 0\n",
    "# I think this is because prokka is unable to find any proteins in these files\n",
    "#busby_phage_123.faa\n",
    "#busby_phage_72.faa\n",
    "#busby_phage_126.faa\n",
    "#busby_phage_91.faa\n",
    "# remove them\n",
    "find . -size 0 -delete\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc0a22f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run vContact2\n",
    "# We can now take the .faa files produced by prokka and use them to build a gene-sharing network!\n",
    "# This will be used to compare the \"relatedness\" of our phage to those previously described by other bee phage papers\n",
    "# We can also choose to run this with a collection of reference phage provided by vContact.\n",
    "    # This is a good thing to do, as it will place our phage (and bee phage more generally) within the context of what we know about phage more generally\n",
    "\n",
    "##### Important #####\n",
    "# Before we can run this script, we need to prepare the all_known_gene2genome.tsv file\n",
    "# This is a file which contains 3 columns \n",
    "    # 1) Protein ID (a unique identified for each protein)\n",
    "    # 2) Contig ID (Phage name -- an indentified which indicates the source [phage] of the gene in question)\n",
    "    # 3) Keywords (Gene annotation -- doesnt actually have to be anything, but can be used for cool analyses!!)\n",
    "# We generate this file using the R script: vcontact2_preprocessing.R\n",
    "    # I did this in R because I am more comfortable with text editing there than in bash\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=vcontact_%j\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 12:00:00\n",
    "#SBATCH --ntasks=16\n",
    "#SBATCH --output=/home/dsbard/bee_phage/reviewer_approach/network/log/vonctact%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "source ~/miniconda3/bin/activate \n",
    "conda activate /home/dsbard/yes/envs/vcontact\n",
    "\n",
    "vcontact2 \\\n",
    "--raw-proteins /home/dsbard/bee_phage/reviewer_approach/network/allseqs.faa \\\n",
    "--proteins-fp /home/dsbard/bee_phage/reviewer_approach/network/gene2genome.tsv \\\n",
    "--output-dir /home/dsbard/bee_phage/reviewer_approach/network/no_ref \\\n",
    "--db None \\ # switch to -db 'ProkaryoticViralRefSeq211-Merged' if you want network with reference phage\n",
    "--rel-mode 'Diamond' \\\n",
    "--pcs-mode MCL \\\n",
    "--vcs-mode ClusterONE \\\n",
    "--c1-bin /home/dsbard/bee_phage/reviewer_approach/scripts/cluster_one-1.0.jar \\\n",
    "--vc-penalty 1 \\\n",
    "-t 16\n",
    "\n",
    "# This gives a collection of files, some of which are more usefull than others\n",
    "# The authors provide a great explination of each component here: \n",
    "    ## https://bitbucket.org/MAVERICLab/vcontact2/src/master/\n",
    "# To make pretty networks from these outputs, see the R script titled: decorate_network_all_phage.R"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
