{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4f99e34",
   "metadata": {},
   "outputs": [],
   "source": [
    "# one of the main reviewer comments is that the viral prediction software we used (virsorter2) is out of date\n",
    "# Instead, reviewers suggest we use genomad\n",
    "# I have install genomad using mamba\n",
    "# lets try it out\n",
    "\n",
    "# First, copy over our virome and total metagenome assemblies to new dir\n",
    "cp /home/dsbard/bee_phage/dls/5kb/virome/assemblies/5kb/* ./\n",
    "cp cp /home/dsbard/bee_phage/dls/5kb/total_metagenome/assemblies/5kb/* ./\n",
    "\n",
    "# run genomad\n",
    "### total metagenome samples\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=Genomad\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=12\n",
    "#SBATCH --output=/home/dsbard/bee_phage/reviewer_approach/genomad_out/log/log/%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "conda activate genomad\n",
    "\n",
    "genomad \\\n",
    "end-to-end \\\n",
    "--cleanup \\\n",
    "--splits 8 \\\n",
    "/home/dsbard/bee_phage/reviewer_approach/assemblies/total_metagenomes/$1 \\\n",
    "/home/dsbard/bee_phage/reviewer_approach/assemblies/genomad_out/total_out \\\n",
    "/home/dsbard/genomad_db/genomad_db\n",
    "\n",
    "# run from /home/dsbard/bee_phage/reviewer_approach/assemblies/total_metagenomes\n",
    "for f in *.fasta\n",
    "do\n",
    "sbatch ../../scripts/genomad_total.sh $f\n",
    "done\n",
    "\n",
    "### Virome samples\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=Genomad\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=12\n",
    "#SBATCH --output=/home/dsbard/bee_phage/reviewer_approach/genomad_out/log/log/%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "conda activate genomad\n",
    "\n",
    "genomad \\\n",
    "end-to-end \\\n",
    "--cleanup \\\n",
    "--splits 8 \\\n",
    "/home/dsbard/bee_phage/reviewer_approach/assemblies/viromes/$1 \\\n",
    "/home/dsbard/bee_phage/reviewer_approach/genomad_out/virome_out \\\n",
    "/home/dsbard/genomad_db/genomad_db\n",
    "\n",
    "# run from /home/dsbard/bee_phage/reviewer_approach/assemblies/viromes\n",
    "for f in *.fasta\n",
    "do\n",
    "sbatch ../../scripts/genomad_virome.sh $f\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2eb0bb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# place IDed phage into new dir\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/genomad_out\n",
    "cd viral_seqs\n",
    "mkdir viral_seqs/total viral_seqs/virome\n",
    "for f in total_out/*summary/*virus.fna; do cp $f ./viral_seqs/total; done\n",
    "for f in virome_out/*summary/*virus.fna; do cp $f ./viral_seqs/virome; done\n",
    "\n",
    "# count how many phage seqs we got out per sample\n",
    "## Total\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/genomad_out/viral_seqs/total\n",
    "\n",
    "# Honey bee\n",
    "for f in *MEL*.fna\n",
    "do\n",
    "sam=${f%%virsorter*}\n",
    "n_phage=$(grep \">\" $f | wc -l)\n",
    "echo $sam,$n_phage\n",
    "done >> honey_total_phage_counts.txt\n",
    "\n",
    "# Bumble bee\n",
    "for f in *IMP*.fna\n",
    "do\n",
    "sam=${f%%virsorter*}\n",
    "n_phage=$(grep \">\" $f | wc -l)\n",
    "echo $sam,$n_phage\n",
    "done >> bumble_total_phage_counts.txt\n",
    "\n",
    "## virome\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/genomad_out/viral_seqs/virome\n",
    "\n",
    "# Honey bee\n",
    "for f in *MEL*.fna\n",
    "do\n",
    "sam=${f%%virsorter*}\n",
    "n_phage=$(grep \">\" $f | wc -l)\n",
    "echo $sam,$n_phage\n",
    "done >> honey_virome_phage_counts.txt\n",
    "\n",
    "# Bumble bee\n",
    "for f in *IMP*.fna\n",
    "do\n",
    "sam=${f%%virsorter*}\n",
    "n_phage=$(grep \">\" $f | wc -l)\n",
    "echo $sam,$n_phage\n",
    "done >> bumble_virome_phage_counts.txt\n",
    "\n",
    "# count number of phage IDed by each sample geNomad\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/genomad_out\n",
    "# virome samples\n",
    "for f in virome_out/*summary/*virus.fna\n",
    "do\n",
    "    sample=${f%%_summary*}\n",
    "    sample_2=${sample#*/}\n",
    "    count=$(grep \">\" $f | wc -l)\n",
    "    echo $sample_2,$count\n",
    "done >> phage_count.csv\n",
    "# Total metagenome samples\n",
    "for f in total_out/*summary/*virus.fna\n",
    "do\n",
    "    sample=${f%%_summary*}\n",
    "    sample_2=${sample#*/}\n",
    "    count=$(grep \">\" $f | wc -l)\n",
    "    echo $sample_2,$count\n",
    "done >> phage_count.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16149998",
   "metadata": {},
   "outputs": [],
   "source": [
    "# copy phage seqs into a new file\n",
    "## all total metagenome phage\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/genomad_out/viral_seqs/total\n",
    "cat *.fna >> all_total_phage.fna\n",
    "# rename them total_0-X\n",
    "conda activate bbduk\n",
    "rename.sh in=all_total_phage.fna \\\n",
    "out=all_total_phage_renamed.fna  \\\n",
    "prefix=total_phage_\n",
    "grep \">\" all_total_phage_renamed.fna | wc -l\n",
    "# 938 phage found\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/genomad_out/viral_seqs/virome\n",
    "cat *.fna >> all_virome_phage.fna\n",
    "rename.sh in=all_virome_phage.fna \\\n",
    "out=all_virome_phage_renamed.fna  \\\n",
    "prefix=virome_phage_\n",
    "grep \">\" all_virome_phage_renamed.fna | wc -l\n",
    "# 2915 phage found\n",
    "\n",
    "# Reviewers also suggested we use drep instead of cdhit (as we had done previously) and that we use checkV to remove low quality phage seqs\n",
    "# now combine virome and total metagenome phage in a new file for dereplication\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/\n",
    "mkdir drep\n",
    "cd drep\n",
    "mkdir log viral_seqs\n",
    "# write new file\n",
    "s\n",
    "\n",
    "# Run checkV\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=checkV\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 24:00:00\n",
    "#SBATCH --ntasks=16\n",
    "#SBATCH --output=/home/dsbard/bee_phage/reviewer_approach/checkv/log/%j.out\n",
    "#SBATCH --partition=bigmemm\n",
    "#SBATCH --mem=120GB\n",
    "\n",
    "conda activate checkv\n",
    "\n",
    "checkv end_to_end \\\n",
    "/home/dsbard/bee_phage/reviewer_approach/checkv/all_phage_seqs.fna \\\n",
    "/home/dsbard/bee_phage/reviewer_approach/checkv/out \\\n",
    "-d /home/dsbard/checkv/checkv-db-v1.5 \\\n",
    "-t 16\n",
    "\n",
    "mkdir trimmed\n",
    "cat out/*.fna >> trimmed/trimmed_seqs.fa\n",
    "\n",
    "# Filter to retain only high quality viral seqs\n",
    "# (this is done in R)\n",
    "# 655 ~high quality~ phage left\n",
    "\n",
    "sed 's/_1 .*$//' trimmed_seqs.fa > cleaned_trimmed_seqs.fa\n",
    "# now lets split em\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/checkv/trimmed\n",
    "cat cleaned_trimmed_seqs.fa | awk '{\n",
    "        if (substr($0, 1, 1)==\">\") {filename=(substr($0,2) \".fa\")}\n",
    "        print $0 > filename\n",
    "}'\n",
    "mv virome_phage* seqs/\n",
    "mv total* seqs/\n",
    "\n",
    "# use list of good phage to pull high quality phage genomes\n",
    "while read line\n",
    "do \n",
    "cp trimmed/seqs/${line}.fa \\\n",
    "/home/dsbard/bee_phage/reviewer_approach/drep/good_seqs\n",
    "done < good_viral_seqs.csv\n",
    "\n",
    "# run drep\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=drep\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 24:00:00\n",
    "#SBATCH --ntasks=12\n",
    "#SBATCH --output=/home/dsbard/bee_phage/reviewer_approach/drep/log/%j.out\n",
    "#SBATCH --partition=bigmemm\n",
    "#SBATCH --mem=120GB\n",
    "\n",
    "\n",
    "conda activate drep\n",
    "\n",
    "dRep dereplicate /home/dsbard/bee_phage/reviewer_approach/drep/dereplicated_phage \\\n",
    "-g /home/dsbard/bee_phage/reviewer_approach/drep/good_seqs/*.fa \\\n",
    "-l 5000 \\\n",
    "-nc .85 \\\n",
    "--ignoreGenomeQuality\n",
    "\n",
    "cd /home/dsbard/bee_phage/reviewer_approach/drep/dereplicated_phage/dereplicated_genomes\n",
    "ls -1 | wc -l\n",
    "# 609 phage out\n",
    "# 655 phage in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6594d867",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate coverage\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=cover_%j\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00 \n",
    "#SBATCH --ntasks=12\n",
    "#SBATCH --output=/home/dsbard/coverm/log/cover%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "# module load deprecated/coverm/0.6.1  \n",
    "# source activate coverm-0.6.1\n",
    "# module load samtools/1.14 \n",
    "source /home/dsbard/miniconda3/bin/activate\n",
    "conda activate /group/nas-10-1-export-4/dsbard/yes/envs/coverm\n",
    "\n",
    "coverm genome -t 12 \\\n",
    "-1 /group/nas-10-1-export-4/dsbard/bee_phage/dls/hybrid/decontam/$1 \\\n",
    "-2 /group/nas-10-1-export-4/dsbard/bee_phage/dls/hybrid/decontam/${1%%R1*}R2.fastq.gz \\\n",
    "--genome-fasta-directory /group/nas-10-1-export-4/dsbard/bee_phage/reviewer_approach/drep/dereplicated_phage/dereplicated_genomes/ \\\n",
    "--genome-fasta-extension .fa \\\n",
    "--min-read-percent-identity 95 \\\n",
    "--min-covered-fraction=75 \\\n",
    "--min-read-aligned-percent 75 \\\n",
    "-m trimmed_mean \\\n",
    "-m covered_fraction \\\n",
    "--output-file /home/dsbard/coverm/out/${1%%_host*}_cover_result.tsv\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
