{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2eaffad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# visualize initial read quality\n",
    "# We will do this with fastqc\n",
    "\n",
    "##### Virome Samples #####\n",
    "# move to new dir\n",
    "cd path/to/raw/virome/read/files\n",
    "# activate conda\n",
    "conda activate fastqc\n",
    "# run fastqc\n",
    "for f in *.gz\n",
    "do\n",
    "fastqc $f\n",
    "done\n",
    "# move files to new dir\n",
    "mkdir fastqc\n",
    "mv *.html fastqc\n",
    "mv *.zip fastqc\n",
    "\n",
    "##### Negative Samples #####\n",
    "# Lets also look at our negative samples\n",
    "cd path/to/raw/negative/read/files\n",
    "conda activate fastqc\n",
    "# run fastqc\n",
    "for f in *.gz\n",
    "do\n",
    "fastqc $f\n",
    "done\n",
    "# move files to new dir\n",
    "mkdir fastqc\n",
    "mv *.html fastqc\n",
    "mv *.zip fastqc\n",
    "\n",
    "##### Total metagenomes #####\n",
    "cd path/to/raw/total_metagenome/read/files\n",
    "conda activate fastqc\n",
    "# Run it\n",
    "for f in *.gz \n",
    "do\n",
    "fastqc $f\n",
    "done\n",
    "# move files to new dir\n",
    "mkdir fastqc\n",
    "mv *.html fastqc\n",
    "mv *.zip fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d00f2ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now lets clean up our seqs\n",
    "# We will do this with...\n",
    "# Trim_galore to remove polyG reads (byproduct of NovaSeq)\n",
    "# And trimmomatic to remove adapters and improve read quality\n",
    "\n",
    "##### NEG #####\n",
    "cd path/to/raw/virome/read/files\n",
    "# Remove Poly-G seqs from read libraries\n",
    "# Make new dir for data\n",
    "mkdir no_Gs\n",
    "# Activate conda env\n",
    "conda activate trim_galore\n",
    "# Run loop\n",
    "for f in *R1*\n",
    "do\n",
    "trim_galore --paired \\\n",
    "$f ${f%%R1*}R2_001.fastq.gz \\\n",
    "--phred33 \\\n",
    "-q 0 \\\n",
    "-a G{100} \\\n",
    "-A G{100} \\\n",
    "-o ./no_Gs\n",
    "done\n",
    "# Remove adapters and quality trim\n",
    "cd no_Gs\n",
    "# Acivate trimmomatic\n",
    "conda activate trimmo\n",
    "# Run loop\n",
    "for f in *R1*.gz\n",
    "do\n",
    "trimmomatic \\\n",
    "PE -threads 4 -phred33 \\\n",
    "$f ${f%%R1*}R2_001_val_2.fq.gz \\\n",
    "../trimmed/${f%%001*}_trimmed.fq.gz \\\n",
    "../trimmed/${f%%001*}_unpaired.fq.gz \\\n",
    "../trimmed/${f%%R1*}R2_trimmed.fq.gz \\\n",
    "../trimmed/${f%%R1*}}R2_unpaired.fq.gz \\\n",
    "ILLUMINACLIP:../adapters/TruSeq3-PE-2.fa:2:30:10:2 \\\n",
    "AVGQUAL:25 MINLEN:40\n",
    "done\n",
    "# move unpaired seqs to their own dir\n",
    "mkdir unpaired\n",
    "mv *unpaired.gz unpaired \n",
    "mkdir paired\n",
    "mv *paired.gz\n",
    "# Run fastqc on trimmed seqs\n",
    "conda activate fastqc\n",
    "for f in *.gz\n",
    "do\n",
    "fastqc $f\n",
    "done\n",
    "# Stash fastqc results in new dir\n",
    "mkdir fastqc\n",
    "mv *.html ./fastqc\n",
    "mv *.zip ./fastqc\n",
    "\n",
    "##### Virome #####\n",
    "cd path/to/raw/virome/read/files\n",
    "# Remove Poly-G seqs from read libraries\n",
    "# Make new dir for data\n",
    "mkdir no_Gs\n",
    "# Activate conda env\n",
    "conda activate trim_galore\n",
    "# Run loop\n",
    "for f in *R1*\n",
    "do\n",
    "trim_galore --paired \\\n",
    "$f ${f%%R1*}R2_001.fastq.gz \\\n",
    "--phred33 \\\n",
    "-q 0 \\\n",
    "-a G{100} \\\n",
    "-A G{100} \\\n",
    "-o ./no_Gs\n",
    "done\n",
    "# Trimmomatic\n",
    "# Remove adapters and quality trim\n",
    "cd no_Gs\n",
    "# Acivate trimmomatic\n",
    "conda activate trimmo\n",
    "# Run loop\n",
    "for f in *R1*.gz\n",
    "do\n",
    "trimmomatic \\\n",
    "PE -threads 4 -phred33 \\\n",
    "$f ${f%%R1*}R2_001_val_2.fq.gz \\\n",
    "../trimmed/${f%%001*}_trimmed.fq.gz \\\n",
    "../trimmed/${f%%001*}_unpaired.fq.gz \\\n",
    "../trimmed/${f%%R1*}R2_trimmed.fq.gz \\\n",
    "../trimmed/${f%%R1*}}R2_unpaired.fq.gz \\\n",
    "ILLUMINACLIP:../adapters/TruSeq3-PE-2.fa:2:30:10:2 \\\n",
    "AVGQUAL:25 MINLEN:40\n",
    "done\n",
    "# move unpaired seqs to their own dir\n",
    "mkdir unpaired\n",
    "mv *unpaired.fq.gz unpaired \n",
    "# Run fastqc on trimmed seqs\n",
    "conda activate fastqc\n",
    "for f in *.gz\n",
    "do\n",
    "fastqc $f\n",
    "done\n",
    "# Stash fastqc results in new dir\n",
    "mkdir fastqc\n",
    "mv *.html ./fastqc\n",
    "mv *.zip ./fastqc\n",
    "\n",
    "##### Total metagenomes #####\n",
    "cd path/to/raw/total_metagenome/read/files\n",
    "# Remove Poly-G seqs from read libraries\n",
    "# Make new dir for data\n",
    "mkdir no_Gs\n",
    "# Activate conda env\n",
    "conda activate trim_galore\n",
    "# Run loop\n",
    "for f in *R1*\n",
    "do\n",
    "trim_galore --paired \\\n",
    "$f ${f%%R1*}R2_001.fastq.gz \\\n",
    "--phred33 \\\n",
    "-q 0 \\\n",
    "-a G{100} \\\n",
    "-A G{100} \\\n",
    "-o ./no_Gs\n",
    "done\n",
    "# Remove adapters and quality trim\n",
    "cd no_Gs\n",
    "# Move .txt files to new dir\n",
    "mkdir reports \n",
    "mv *.txt reports\n",
    "# Acivate trimmomatic\n",
    "conda activate trimmo\n",
    "# Run loop\n",
    "for f in *R1*.gz\n",
    "do\n",
    "trimmomatic \\\n",
    "PE -threads 4 -phred33 \\\n",
    "$f ${f%%R1*}R2_001_val_2.fq.gz \\\n",
    "../trimmed/${f%%001*}_trimmed.fq.gz \\\n",
    "../trimmed/${f%%001*}_unpaired.fq.gz \\\n",
    "../trimmed/${f%%R1*}R2_trimmed.fq.gz \\\n",
    "../trimmed/${f%%R1*}}R2_unpaired.fq.gz \\\n",
    "ILLUMINACLIP:../adapters/TruSeq3-PE-2.fa:2:30:10:2 \\\n",
    "AVGQUAL:25 MINLEN:40\n",
    "done\n",
    "# move unpaired seqs to their own dir\n",
    "cd trimmed\n",
    "mkdir unpaired\n",
    "mv *unpaired.fq.gz unpaired \n",
    "# Run fastqc on trimmed seqs\n",
    "conda activate fastqc\n",
    "for f in *.gz\n",
    "do\n",
    "fastqc $f\n",
    "done\n",
    "# Stash fastqc results in new dir\n",
    "mkdir fastqc\n",
    "mv *.html ./fastqc\n",
    "mv *.zip ./fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d57bc6c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# We now transfer these semi cleaned reads to our HPC for the next steps\n",
    "\n",
    "##### NEG #####\n",
    "rsync -v path/to/neg/trimmed/*.gz \\\n",
    "dsbard@farm.cse.ucdavis.edu:/home/dsbard/bee_phage/dls/neg/trimmed_reads\n",
    "\n",
    "##### virome #####\n",
    "rsync -v path/to/virome/trimmed/*.gz \\\n",
    "dsbard@farm.cse.ucdavis.edu:/home/dsbard/bee_phage/dls/virome/trimmed_reads\n",
    "    \n",
    "##### total metagenomes #####\n",
    "rsync -v path/to/total_metagenomes/trimmed/*.gz \\\n",
    "dsbard@farm.cse.ucdavis.edu:/home/dsbard/bee_phage/dls/total_metagenomes/trimmed_reads"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
