{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a950eae0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# So we have cleaned our reads fairly well up until this point\n",
    "    # Removed polyG seqs\n",
    "    # Removed adapter seqs\n",
    "    # Quality filtered\n",
    "# But now we want to remove any potential contamination\n",
    "# This includes...\n",
    "    # Host bee seqs\n",
    "    # Potential extraction contamination\n",
    "# To do this, we will \n",
    "    # 1) Construct scaffolds from our negative reads\n",
    "        # These is our potential extraction contamination stuff\n",
    "    # 2) Next, we will add host bee seqs \n",
    "        # B. imp and A. mel genomes\n",
    "    # 3) We will then align all of our reads against this database of potential contamination \n",
    "    # 4) Any reads which map to those negative contams will be removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc3565d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Build scaffolds from negative reads\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=Assemblies\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00 \n",
    "#SBATCH --ntasks=8\n",
    "#SBATCH --output=/path/to/neg/assemebly/log_files/%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "module load spades\n",
    "\n",
    "spades.py \\\n",
    "-o /path/to/neg/assemebly/${1%%_S*} \\\n",
    "-1 $1 \\\n",
    "-2 ${1%%R1*}R2_trimmed.fq.gz \\\n",
    "-k 21,33,55,77,99,127 \\\n",
    "--meta \\\n",
    "-m 120 \\\n",
    "-t 8\n",
    "\n",
    "# Run this shell script as a loop\n",
    "cd path/to/neg/trimmed_reads\n",
    "for f in *R1*.gz\n",
    "do\n",
    "sbatch ../scripts/assemble.sh $f\n",
    "done\n",
    "\n",
    "# Collect output in new file \n",
    "cd path/to/neg/assemblies\n",
    "mkdir scaffolds\n",
    "cat *NEG/scaffolds.fasta >> scaffolds/negative_seqs.fasta\n",
    "\n",
    "# Lets move this negative_seqs file to a new dir\n",
    "cd path/to/neg\n",
    "mkdir negative_host_contam\n",
    "cp path/to/neg/assemblies/scaffolds/negative_seqs.fasta ./negative_host_contam\n",
    "\n",
    "# Now we throw in the Bombus impatiens and Apis mellifera genomes\n",
    "scp path/to/downloaded/host/bee/genomes/*.fna \\\n",
    "dsbard@farm.cse.ucdavis.edu:/path/to/neg/negative_host_contam\n",
    "# Combine bee genomes and negative contam into one big file\n",
    "cd path/to/neg/negative_host_contam\n",
    "cat negative_host_contam \\\n",
    "GCA_000002195.1_Amel_4.5_genomic.fna \\\n",
    "GCF_000188095.3_BIMP_2.2_genomic.fna \\\n",
    ">> contaminating_seqs.fasta\n",
    "\n",
    "# Make a local database from this contam file using bowtie\n",
    "srun -p bigmemm --time=1:00:00 --nodes=1 \\\n",
    "--cpus-per-task 1 --mem 5 --pty /bin/bash\n",
    "conda activate bowtie2\n",
    "cd path/to/neg/negative_host_contam\n",
    "bowtie2-build contaminating_seqs.fasta negative_db\n",
    "mkdir db\n",
    "mv *.bt2 db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1fc983fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now we actually use Bowtie2 to remove contamination from our read libraries\n",
    "\n",
    "##### Virome #####\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=decontam\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=12\n",
    "#SBATCH --output=/home/dsbard/bee_phage/dls/virome/decontam/log/decontam%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "conda activate bowtie2\n",
    "\n",
    "bowtie2 -p 1 -x path/to/neg/negative_host_contam/db/negative_db \\\n",
    "  -1 path/to/virome/trimmed_reads/$1 \\\n",
    "  -2 path/to/virome/trimmed_reads/${1%%R1*}R2_trimmed.fq.gz \\\n",
    "  --un-conc-gz \\\n",
    "  path/to/virome/decontam/${1%%_S*}_host_removed \\\n",
    "  > path/to/virome/decontam/${1%%_S*}_host_removed_mapped_and_unmapped.sam\n",
    "\n",
    "# run shell script from dir with trimmer reads\n",
    "for f in *R1*\n",
    "do\n",
    "sbatch ../scripts/decontam.sh $f\n",
    "done\n",
    "\n",
    "# Then rename outputs\n",
    "for f in *removed.1\n",
    "do\n",
    "mv $f ${f%%.1}_R1.fastq.gz\n",
    "done\n",
    "for f in *removed.2\n",
    "do\n",
    "mv $f ${f%%.2}_R2.fastq.gz\n",
    "done\n",
    "\n",
    "##### Total metagenome #####\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=decontam\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=12\n",
    "#SBATCH --output=/home/dsbard/bee_phage/dls/total_metagenomes/decontam/log/decontam%j.out\n",
    "#SBATCH --partition=bigmemm\n",
    "\n",
    "conda activate bowtie2\n",
    "\n",
    "bowtie2 -p 1 -x path/to/neg/negative_host_contam/db/negative_db \\\n",
    "  -1 path/to/total_metagenomes/trimmed_reads/$1 \\\n",
    "  -2 path/to/total_metagenomes/trimmed_reads/${1%%R1*}R2_trimmed.fq.gz \\\n",
    "  --un-conc-gz \\\n",
    "  path/to/total_metagenomes/decontam/${1%%_S*}_host_removed \\\n",
    "  > path/to/total_metagenomes/decontam/${1%%_S*}_host_removed_mapped_and_unmapped.sam\n",
    "\n",
    "# run shell script from dir with trimmer reads\n",
    "for f in *R1*\n",
    "do\n",
    "sbatch ../scripts/decontam.sh $f\n",
    "done\n",
    "\n",
    "# Then rename outputs to \n",
    "for f in *removed.1\n",
    "do\n",
    "mv $f ${f%%.1}_R1.fastq.gz\n",
    "done\n",
    "for f in *removed.2\n",
    "do\n",
    "mv $f ${f%%.2}_R2.fastq.gz\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
