{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fa16339",
   "metadata": {},
   "outputs": [],
   "source": [
    "# The goal of this script is to copnstruct Metagenome Assemebled Genomes (MAGs)\n",
    "# We will be building these MAGs from our total metagenome data\n",
    "    # Specifically, the assemebelies we built from total metagenome reads using MetaSpades\n",
    "# To construct out MAGs we will....\n",
    "    # Use MetaBat to bin assembelies into \"crude\" MAGs\n",
    "    # We will then use CheckM to check the quality of our MAGs\n",
    "    # Drep is then used to remove redundant MAGs from our high-quality MAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13e45561",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First step, use MetaBat to bin assembelies\n",
    "# However, MetaBat requires a .bam file describing read depth and alignment location\n",
    "# It also wants this .bam file to be sorted\n",
    "# So, to use MetaBat, we will construct these .bam files by...\n",
    "    # 1) mapping reads to assembelies with bowtie2\n",
    "    # 2) converting the output .sam files into .bam\n",
    "    # 3) sorting the .bam files with samtools\n",
    "    # 4) run MetaBat!!\n",
    "\n",
    "# Bowtie step (making .sam file)\n",
    "# Start by building bowtie2 indexes\n",
    "# Make empty dir\n",
    "cd /path/to/MAG_construction/assemblies/scaffolds\n",
    "for f in *.fa\n",
    "do\n",
    "mkdir /path/to/MAG_construction/binning/${f%%.fa*}\n",
    "done\n",
    "\n",
    "# Then run make bowtie2 db\n",
    "conda activate bowtie2\n",
    "for f in *.fa\n",
    "do\n",
    "bowtie2-build \\\n",
    "/path/to/MAG_construction/assemblies/scaffolds/$f \\\n",
    "/path/to/MAG_construction/binning/${f%%.fa*}/${f%%.fa*}_db\n",
    "done\n",
    "\n",
    "# Now we will run the actual mapping\n",
    "cd /path/to/MAG_construction/binning\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=make_SAM\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=12\n",
    "#SBATCH --output=/path/to/MAG_construction/binning/log/%j.out\n",
    "#SBATCH --partition=med\n",
    "\n",
    "conda activate bowtie2\n",
    "\n",
    "bowtie2 -x /path/to/MAG_construction/binning/${1%%_host*}/${1%%_host*}_db \\\n",
    "-1 /path/to/trimmed/and/cleaned/reads/$1 \\\n",
    "-2 /path/to/trimmed/and/cleaned/reads/${1%%R1*}R2.fastq.gz \\\n",
    "-S /path/to/MAG_construction/binning/${1%%_host*}.sam \\\n",
    "--very-sensitive-local\n",
    "\n",
    "# Run it\n",
    "conda activate bowtie2\n",
    "cd /path/to/trimmed/and/cleaned/reads\n",
    "for f in *R1*.gz\n",
    "do\n",
    "sbatch ../scripts/make_SAM.sh $f\n",
    "done\n",
    "\n",
    "# Then move the outputs to a new dir\n",
    "mkdir sam_files\n",
    "mv *.sam sam_files/\n",
    "# Next, we have to convert these .sam files into .bam files\n",
    "# We will do this with samtools\n",
    "conda activate samtools\n",
    "for f in *.sam\n",
    "do\n",
    "samtools view -S -b $f > ${f%%.sam}.bam\n",
    "done\n",
    "# Next, we have to sort\n",
    "for f in *.bam\n",
    "do\n",
    "samtools sort $f -o ${f%%.bam}_sorted.bam\n",
    "done\n",
    "\n",
    "# Now we are ready to run MetaBat\n",
    "    # Note, this was done on an HPC in an srun environment\n",
    "conda activate metabat\n",
    "for f in *.fa\n",
    "do\n",
    "runMetaBat.sh \\\n",
    "/path/to/MAG_construction/assemblies/scaffolds/$f \\\n",
    "/path/to/MAG_construction/binning/sam_files/sorted_bam/${f%%.fa}_sorted.bam\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dd8ab35",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now we want to check the quality of our bins/MAGs\n",
    "# We will do this through checkM\n",
    "\n",
    "# Copy and rename bins to new dir named bin_seqs\n",
    "cd /path/to/MAG_construction/binning/bins\n",
    "for f in */*\n",
    "do\n",
    "sam=${f%%.fa*}\n",
    "bin=${f##*/}\n",
    "cp $f ./bin_seqs/${sam}_${bin}\n",
    "done\n",
    "\n",
    "# Load checkm\n",
    "    # Note, this was done on an HPC in an srun env with a lot of ram\n",
    "checkm lineage_wf -t 8 -x fa \\\n",
    "/path/to/MAG_construction/binning/bins/bin_seqs \\\n",
    "./checkm_out\n",
    "\n",
    "# After some processing in R\n",
    "# (selecting for Bins with <= 2% contam and >= 90% completion)\n",
    "# copy high quality bins using list\n",
    "cd /path/to/MAG_construction/binning/bins/bin_seqs/checkm_out\n",
    "\n",
    "while read f; do\n",
    "cp ../${f}.fa ~/bee_phage/dls/total_metagenomes/checkM\n",
    "done < high_quality_bins.csv\n",
    "\n",
    "# Move to that dir\n",
    "cd ~/bee_phage/dls/total_metagenomes/checkM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c94f2e7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Finally, Dereplicate with dRep\n",
    "cd /path/to/MAG_construction\n",
    "mkdir drep\n",
    "# Uses checkM, so will have to be done with a lot of RAM too\n",
    "dRep dereplicate ./ -g \\\n",
    "/path/to/MAG_construction/binning/bins/bin_seqs/checkM/*.fa\n",
    "# After dereplicating we are left with 23 MAGS\n",
    "# 12 from honey bees\n",
    "# 11 from bumble bees"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
